modules/configuration/src/main/java/org/apache/ignite/configuration/storage/ConfigurationStorage.java

```diff
@@ -44,6 +44,8 @@
      * Add listener to the storage that notifies of data changes.
      * @param listener Listener.
      */
+    // TODO: seems that it's not needed to have an ability to set several listeners to storage, as far as only one is responsible
```

Comment:
Please assign ticket to every TODO that yo put in the code, otherwise we'll lose them

Reply:
Let's remove addListener() and use constructor instead of it.

Reply:
This is not obvious to implement as far as DistributedConfigurationStorage instance is created before ConfigurationChanger, but we need to set `org.apache.ignite.configuration.ConfigurationChanger#updateFromListener` to `DistributedConfigurationStorage` at some point, so the constructor doesn't fit here. I would do cush changes in a different ticket

Reply:
created a ticket https://issues.apache.org/jira/browse/IGNITE-14689

Reply:
I'm ok with doing it in separate ticket.